### DISCUSSION TITLE
Understanding RAM requirement by Milvus

### DISCUSSION THREAD

**Discussion by NehaTamore** [milvus-io/milvus#22569] [Q&A and General discussion] [answered]
I'm fairly new to using Milvus. and planning to build a system of about 224D, 300M vectors. I have checked milvus sizing tool, according to which system requires 500GB of overall memory with FLAT index and 250.4GB of raw file size.

My questions are as follows:
1. What operation consume 500GB of memory? index creation?
2. Post index creation(suppose using IVF_FLAT with nlist=30, n_probe=1), when we load the collection, index file and respective partition vectors(max 10M *n_probe) shall be loaded into memory. Is this correct?

In that case, post first index creation, a static system shall not require entirely 500GB of memory. Thus could we scale down the system post index creation?

**Comment by xiaofan-luan** [comment, marked as answer]
> I'm fairly new to using Milvus. and planning to build a system of about 224D, 300M vectors. I have checked milvus sizing tool, according to which system requires 500GB of overall memory with FLAT index and 250.4GB of raw file size.
> 
> My questions are as follows:
> 
> 1. What operation consume 500GB of memory? index creation?
> 2. Post index creation(suppose using IVF_FLAT with nlist=30, n_probe=1), when we load the collection, index file and respective partition vectors(max 10M *n_probe) shall be loaded into memory. Is this correct?
> 
> In that case, post first index creation, a static system shall not require entirely 500GB of memory. Thus could we scale down the system post index creation?

Hi Neha,  most memory consumption part would be index serving, most of the index required to load index into memory.
For my experience, querynode requires slightly more than index size(So let's say 300GB), indexNode requires 8c 32G(and it can be released once index build done). DataNode requires 8G memory, proxy, coordinators requires 1-2G memory.
